cityofaustin/atd-moped
# Update component create form with React Hook Form (#3)

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/10401

~@johnclary Tagging you for review on the UI flow and to make sure it covers what we are looking for 🙏~  ✅  - Thanks, John!
Probably needs some tweaks to the form output once we start mutating 

I took a quick stab at testing the two hooks that create the Autocomplete options, but I rapidly realized that was a good job for React Testing Library. I'll come back to that down the line in this epic.

## Testing
**URL to test:** 

https://md-10401-draft-component-form--atd-moped-main.netlify.app/moped/ or local

**Steps to test:**
1. Go to a new or existing project and click on the **Map** tab
2. Click the **New Component** button
3. Open the **Component Type** dropdown and you should see options that have types and subtypes (if available)
4. Test out clearing a selected option and typing for autocomplete results
5. Select component type **Bike Box**
6. The **Subcomponents** field should be disabled since there are no subcomponents for this type
7. Clear out **Bike Box** and select **Signal PHB**
8. The **Subcomponents** field should have three options and you should be able to select multiple
9. Add test text to the **Description** box
10. Click **Continue**
11. Select a point from the map and click **Save**
12. Your new component should be saved in state

---
#### Ship list
